packages/yew/src/virtual_dom/mod.rs

```diff
@@ -495,6 +499,12 @@ pub(crate) trait VDiff {
     /// Remove self from parent.
     fn detach(&mut self, parent: &Element);
 
+    /// Move elements from one parent to another parent.
+    /// This is currently only used by `VSuspense` to preserve component state without detaching
+    /// (which destroys component state).
+    /// Prefer `detach` then apply if possible.
+    fn shift(&self, previous_parent: &Element, next_parent: &Element, next_sibling: NodeRef);
+
```

Comment:
Can you comment why this can't reuse `VNode::move_before`? There is a slight difference in that the `next_sibling` is a `NodeRef` instead of the dereferenced `Option<&Node>` but I don't immediately see how that matters.

Reply:
`move_before` assumes previous position and next position are under the same parent and hence does not attempt to update `ComponentState` where as `shift` has a different parent and hence is updated via `push_component_update` for VComp.